src/wallet/walletdb.cpp

```diff
@@ -400,7 +400,7 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             if (wtx.nOrderPos == -1)
                 wss.fAnyUnordered = true;
 
-            pwallet->AddToWallet(wtx, true, NULL);
+            pwallet->AddToWallet(wtx, true);
```

Comment:
If you have 10'000 wallet transaction, this would now result in 10'000 x `CWalletDB walletdb(strWalletFile);`.
What about the performance impacts?